src/reducers/starWarsReducer.js

```diff
@@ -1,13 +1,32 @@
-import /* we need our action types here*/ "../actions";
+import {FETCH_START,FETCH_SUCCESS,FETCH_FAIL} from '../actions';
+
 const initialState = {
-  characters: []
+  characters: [],
+  fetching: false,
   // Array characters, Boolean fetching, null error.
 };
+
 export const charsReducer = (state = initialState, action) => {
   switch (action.type) {
     // Fill me in with the important reducers
     // action types should be FETCHING, SUCCESS and FAILURE
     // your switch statement should handle all of these cases.
+    case FETCH_START:
+      return {
+        ...state,
+        fetching:true,
+      };
+    case FETCH_SUCCESS:
+      return {
+        ...state,
+        characters: [...state.characters, ...action.payload],
```

Comment:
perfect! spreading in the payload like this is more pure, simply because the characters held on state might be altered in future functionalities of the app, and this would handle such a case